src/containerapp/azext_containerapp/_archive_utils.py

```diff
@@ -54,16 +53,10 @@ def upload_source_code(cmd, client,
     if not upload_url:
         raise CLIInternalError("Failed to get a SAS URL to upload context.")
 
-    account_name, endpoint_suffix, container_name, blob_name, sas_token = get_blob_info(upload_url)
-    BlockBlobService = get_sdk(cmd.cli_ctx, ResourceType.DATA_STORAGE, 'blob#BlockBlobService')
-    BlockBlobService(account_name=account_name,
-                     sas_token=sas_token,
-                     endpoint_suffix=endpoint_suffix,
-                     # Increase socket timeout from default of 20s for clients with slow network connection.
-                     socket_timeout=300).create_blob_from_path(
-                         container_name=container_name,
-                         blob_name=blob_name,
-                         file_path=tar_file_path)
+    BlobClient = get_sdk(cmd.cli_ctx, ResourceType.DATA_STORAGE_BLOB, '_blob_client#BlobClient')
```

Comment:
Could you please clarify where we're actually moving from V1 to V2? Here the code looks like what we need to update after we moved to the newer SDK, but I cannot find where the actual "Storage SDK update" happened. What am I missing?

Reply:
For resource type ResourceType.DATA_STORAGE_BLOB and DATA_STORAGE, they will correspond to different SDK.
https://github.com/Azure/azure-cli/blob/82358037c5af4a43c46facfcff054100a730c678/src/azure-cli-core/azure/cli/core/profiles/_shared.py#LL121-L126

Reply:
Yep, sorry, it's not obvious just from changing the `ResourceType` property, but the snippet that Zunli linked above is exactly what the Storage folks sent me when recommending to upgrade to their V2 SDK.

So we needed to change the `ResourceType` property so the `get_sdk` function targets the new SDK, and then we create a `BlobClient` object defined in the V2 Storage's `_blob_client.py` file.